bundles/org.openhab.binding.ism8/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="ism8"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="device" extensible="switch, number">
```

Comment:
Adding extensible like this is not enough. You need to specify the channel-type with these ids. See my other comment. I also would recommend to not have a channel property write, but to specify 2 different channel types, a writeable and a readonly. That way the UI can make a channel really readonly. With the channel option write the ui will stil show an editable ui element. With a readonly channel type this won't be the case. In the code you won't need to make changes for this, except for not having to check for write.

Reply:
Thanks for this info. I looked into the options and created the 4 different channel types. Additionally I added options for the type for each channel. Now it's becoming really easy to configure the binding as each type is selectable as defined in the manual.
The additional parameter used previously (`write=true`) is also removed, as the `...-readonly` types should handle it. Unfortunately does this not totally avoid any write command to the heater. Switching the modes is done via `Selection` and it looks this control is ignoring the readonly-state. It's not really harming as the ISM is simply not accepting any write command, but with the previous solution the state of the item didn't update. Do I still miss something or is the `Selection` control not supporting it?